### PR TITLE
Add data size score metric

### PR DESCRIPTION
**Pull request by meljbruno**
Performs statistical analysis on origin connection data sizes and adds a data size score (ds_score) metric to the beacon analysis output.

### CODE REVIEW

File: analysis/structure/uconn.go [hunk 1]
```diff
@@ -88,6 +88,9 @@ func getUniqueConnectionsScript(sysCfg *config.SystemConfig) (string, string, []
 						}},
 					}},
 				}},
+				{"orig_bytes", bson.D{
```
Comment (Zalgo2462) [inline analysis/structure/uconn.go]: This should be removed

File: datatypes/structure/structure.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ type (
 		Dst             string        `bson:"dst"`
 		LocalSrc        bool          `bson:"local_src"`
 		LocalDst        bool          `bson:"local_dst"`
-		TotalBytes      int           `bson:"total_bytes"`
```
Comment (Zalgo2462) [inline datatypes/structure/structure.go]: Undo this change. We need the total bytes associated with unique connections.

File: analysis/structure/uconn.go [hunk 3]
```diff
@@ -110,6 +113,7 @@ func getUniqueConnectionsScript(sysCfg *config.SystemConfig) (string, string, []
 				{"local_src", 1},
 				{"local_dst", 1},
 				{"total_bytes", 1},
+				{"orig_bytes", 1},
```
Comment (Zalgo2462) [inline analysis/structure/uconn.go]: remove this as well

File: analysis/structure/uconn.go (hunk 1 above)
Comment (meljbruno) [inline analysis/structure/uconn.go]: Fixed in 0ea6d0536b6373fcb75f0d64cf7cb28ee0a6f2e8.

File: analysis/structure/uconn.go (hunk 3 above)
Comment (meljbruno) [inline analysis/structure/uconn.go]: Fixed in 0ea6d0536b6373fcb75f0d64cf7cb28ee0a6f2e8.

File: datatypes/structure/structure.go (hunk 2 above)
Comment (meljbruno) [inline datatypes/structure/structure.go]: Fixed in 0ea6d0536b6373fcb75f0d64cf7cb28ee0a6f2e8.

File: datatypes/structure/structure.go [hunk 4]
```diff
@@ -18,6 +18,7 @@ type (
 		Dst             string        `bson:"dst"`
 		LocalSrc        bool          `bson:"local_src"`
 		LocalDst        bool          `bson:"local_dst"`
+		TotalBytes      int           `bson:"total_bytes"`
 		OriginBytes     int64         `bson:"origin_bytes"`
```
Comment (Zalgo2462) [inline datatypes/structure/structure.go]: We'll want to remove this line

Comment (Zalgo2462) [inline datatypes/structure/structure.go]: Specifically the origin bytes line

**Comment by meljbruno** [comment]
Extraneous line in structure.go fixed, score metrics merged, and new data size metrics added in commit 8ee957a9b4368ae355852e9c53097f08dd9d82a5.

**Comment by Zalgo2462** [comment]
Look's good! We should verify that previously highly scored beacons are still highly scored.

**Comment by Zalgo2462** [comment]
Can you please rebase this off of dev before we merge it in? Thanks!

**Comment by meljbruno** [comment]
Debasing dev should be all set.

**Comment by meljbruno** [comment]
Added the data size metrics to HTML beacon reports, removed "TS" from score variable names and headers in commit 3caa12d1eafae6dd77be9d523d49b970ab85d393.

**Comment by ethack** [comment]
@Zalgo2462 Is this good to be merged?

**Comment by Zalgo2462** [comment]
I'm currently getting an error when running html-report.
`html/template:beacon.html: "<" in attribute name: " Mode</th><th>Intvl. Mode Count<"`

**Comment by Zalgo2462** [comment]
`size_mode` and `size_mode_count` appear to be displaying the same information in the html report.

File: analysis/beacon/beacon.go [hunk 5]
```diff
@@ -284,24 +307,40 @@ func (t *Beacon) analyze() {
...
+		}		
+		
 		gamma := duration
+		//smaller data sizes receive a higher score
+		zeta := 1.0 - (float64(ds_mode) / 65535.0)
 
-		//in order of ascending importance: skew, duration, dispersion
-		output.TS_score = (alpha + beta + gamma) / 3.0
+		//in order of ascending importance: timestamp skew, timestamp duration,
+		//timestamp dispersion, size skew, size duration, size weight
```
Comment (Zalgo2462) [inline analysis/beacon/beacon.go]: s/size duration/size dispersion/

**Comment by Zalgo2462** [comment]
Also, while this code is open for changes, I'd suggest that we remove the TS_duration metric. I've found better results without it. Also, the theory behind TS_duration doesn't make much sense. The metric looks at the observation period (which is usually 1 day), and looks at the time between the first and last connection between the two hosts. It divides this span of time by the observation period. Basically, it tests if a beacon filled the observation period.

**Comment by meljbruno** [comment]
Fixed both issues in commit 633215cf5041dbd0a7fd4ba31e352c782dc04d1e. Opening a new pull request for the removal of the TS_duration metric.
